content/blog/integrating-istio-and-spire.md

```diff
@@ -0,0 +1,463 @@
+```
+
+For details reach out to **[ISTIO download page](https://istio.io/latest/docs/setup/getting-started/#download)**.
+
+cd into the Istio directory and set the path by command:
+
+```shellsession
+export PATH=$PWD/bin:$PATH
+```
+
+After exporting get out of directory.
+
+```shellsession
+cd ..
+```
+
+**Note:** In the future, a case might occur when your cluster does not recognize istioctl. In this case, export the path again after getting into istio directory.
+
+#### **Install Istio with patches:**
+
+After deploying SPIRE into your environment and verifying that all deployments are in Ready state, install Istio with custom patches for the Ingress-gateway as well as for Istio-proxy.
+
+Get the istio-spire-config patch using **[this link](https://raw.githubusercontent.com/cxteamtrials/caas-trials-content/main/services/istio/release-1.17/spire/spire-patch.yaml)** and copy that patch into your cluster. Install that patch using following command.
+
+```shellsession
+istioctl install -f istio-spire-config.yaml
+```
+
+This will share the spiffe-csi-driver with the Ingress Gateway and the sidecars that are going to be injected on workload pods, granting them access to the SPIRE Agent’s UNIX Domain Socket.
+
+* #### Patching Istio-Ingress gateways
+
+  If you receive the error shown below, your ingress-gateway is not patched yet and is not being registered onto the server.
+
+  ![](/img/patch-error-ingress.jpg)
+
+   For patching, the first step is to get and apply one of SPIRE controller manager’s [CRD(Custom Resource Definition)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) ClusterSPIFFEID. It is a cluster-wide resource used to register workloads with SPIRE. The ClusterSPIFFEID can target all workloads in the cluster or can be optionally scoped to specific pods or namespaces via label selectors.
+
+  Create a ClusterSPIFFEID CRD to generate registration entries in SPIRE server for all workloads with the label **`spiffe.io/spire-managed-identity: true`**.
+
+  Get the ClusterSPIFFEID used by us for this demo using **[this link](https://raw.githubusercontent.com/cxteamtrials/caas-trials-content/main/services/spire/clusterspiffeid-example.yaml)**, copy that into your cluster, and apply it.
+
+  ```shellsession
+  kubectl apply -f cluster-spiffeID-crd.yaml
+  ```
+
+  **Note:** You can create your own custom clusterSPIFFEID CRD with your own match label and own selector. For now, we have created simple CRD with one pod selector and one match label.
+
+  Now simply patch the ingress-gateway with spiffe.io/spire managed-identity: true label.
+
+   This will register your ingress-gateway pod into the server.  
+
+  ```shellsession
+  kubectl patch deployment istio-ingressgateway -n istio-system -p '{"spec":{"template":{"metadata":{"labels":{"spiffe.io/spire-managed-identity": "true"}}}}}'
+  ```
+
+  After patching, confirm the working of your ingress-gateway pod, istiod, and all their containers.
```

Comment:
Consider rephrasing to:
After patching, confirm that your ingress-gateway pod, istiod, and all their containers work.